src/pkg/oidc/secret.go

```diff
@@ -86,6 +88,9 @@ var m SecretManager = &defaultManager{
 func (dm *defaultManager) VerifySecret(ctx context.Context, username string, secret string) (*UserInfo, error) {
 	log.Debugf("Verifying the secret for user: %s", username)
 	oidcUser, err := dm.metaDao.GetByUsername(ctx, username)
+	if errors.Is(err, orm.ErrNoRows) {
```

Comment:
Can we move the check into the DAO? This would allow other callers to avoid duplicating the same logic.